zulip_bots/zulip_bots/bots/susi/susi.py

```diff
@@ -41,9 +41,38 @@ def handle_message(self, message: Dict[str, str], bot_handler: Any) -> None:
             return
         reply = requests.get("https://api.susi.ai/susi/chat.json", params=dict(q=msg))
         try:
-            answer = reply.json()['answers'][0]['actions'][0]['expression']
-        except Exception:
+            data = reply.json()
+            actions = data['answers'][0]['actions']
+            answer = ""
+            for action in actions:
+                if action['type'] == 'answer':
+                    answer += action['expression'] + '\n'
+                elif action['type'] == 'anchor':
+                    text = action['text']
+                    link = action['link']
+                    answer += '[' + text + '](' + link+')' + '\n'
+                elif action['type'] == 'table':
+                    table_data = data['answers'][0]['data']
+                    msg = ""
+                    columns = list(action['columns'].keys())
+                    # using markdown make a table
+                    for heading in list(action['columns'].values()):
+                        answer += "| " + heading + " "
+                    answer += "\n"
+                    for heading in columns:
+                        answer += "|" + "------" + " "
+                    answer += "\n"
+                    for item in table_data:
+                        if item[columns[0]]:
+                            column_first = item[columns[0]] if (type(item[columns[0]]) is str) else item[columns[0]][0]
+                            column_second = item[columns[1]] if (type(item[columns[1]]) is str) else item[columns[1]][0]
+                            column_third = item[columns[2]] if (type(item[columns[2]]) is str) else item[columns[2]][0]
+                            msg = "| " + column_first + " | " + column_second + " | " + column_third + "| "
+                            answer = answer + msg + "\n"
+                    answer += "\n"
+        except Exception as e:
             answer = "I don't understand. Can you rephrase?"
+            print(e)
```

Comment:
is this print() intentional ?

Reply:
Yes, it was for debugging. Removed it now.